backend/src/managers/PollManager.ts

```diff
@@ -0,0 +1,186 @@
+import { RowDataPacket } from 'mysql2'
+
+import { PollEntity, PollSettingsEntity } from '../api/types/entities/PollEntity'
+import PollRepository from '../db/repositories/PollRepository'
+
+interface PollRecord extends RowDataPacket {
+  poll_id: number
+  author_id: number
+  site_id: number
+  question: string
+  options: string
+  settings: string
+  expires_at: string | null
+  created_at: string
+  [key: `opt${number}`]: number
+}
+
+interface PollRecordParsed extends Omit<PollRecord, 'options' | 'settings'> {
+  options: string[]
+  settings: PollSettingsEntity
+}
+
+export default class PollManager {
+  private pollRepository: PollRepository
+
+  constructor(pollRepository: PollRepository) {
+    this.pollRepository = pollRepository
+  }
+
+  private async getPollWithVotes(pollId: number, userId?: number): Promise<PollEntity | null> {
+    const poll = await this.pollRepository.getPoll(pollId)
+    if (!poll) {
+      return null
+    }
+
+    const parsedPoll = poll as unknown as PollRecordParsed
+    const options = parsedPoll.options
+    const settings = parsedPoll.settings
+    const userVotes = userId ? await this.pollRepository.getUserVotes(pollId, userId) : undefined
+
+    const totalVotes = Array.from({ length: 32 }, (_, i) => poll[`opt${i}`] || 0).reduce((sum, count) => sum + count, 0)
+
+    const optionsWithVoters = await Promise.all(
+      options.map(async (text: string, index: number) => {
+        const voters = await this.pollRepository.getOptionVoters(pollId, index)
```

Comment:
this is extremely inefficient

individual votes are not needed in most cases, unless somebody requests them from the UI / API

please extract this as a separate endpoint